### PR TITLE
[release/7.0][wasm] SIMD related build fixes

### PR DESCRIPTION
**Pull request by radical**
- rename `$(WasmSIMD)` -> `$(WasmEnableSIMD)`
- Require the workload when the property is set
- Add tests for the various cases

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -66,7 +66,7 @@
                                               Defaults to false.
       - $(WasmAotProfilePath)               - Path to an AOT profile file.
       - $(WasmExceptionHandling)            - Enable support for the WASM Exception Handling feature.
-      - $(WasmSIMD)                         - Enable support for the WASM SIMD feature.
+      - $(WasmEnableSIMD)                         - Enable support for the WASM SIMD feature.
```
Comment (radekdoulik) [inline src/mono/wasm/build/WasmApp.targets:69]: I wanted to keep it similar to other feature props, like the `WasmExceptionHandling` above. Shouldn't we then change that too?

**Review by radekdoulik — APPROVED** [review]
Besides the renaming question, it LGTM. It is great to have more scenarios tested.

**Comment by carlossanlop** [comment]
Ping @marek-safar, need an approval.
I assume failures are unrelated (no WASM hits).

**Review by marek-safar — APPROVED** [review]

**Comment by carlossanlop** [comment]
@radical there's a merge conflict. Can you please address it?

**Comment by radical** [comment]
This is ready for merging.

**Pull request merged**